ttnn/cpp/ttnn/operations/pool/upsample/upsample.hpp

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/tensor/types.hpp"
+#include "ttnn/operations/core/core.hpp"
+
+#include "ttnn/run_operation.hpp"
+
+#include "device/upsample_op.hpp"
+#include "ttnn/operations/pool/upsample/device/upsample_op.hpp"
+
+namespace ttnn {
+namespace operations {
+namespace upsample {
+
+struct ExecuteUpSample {
+    static ttnn::Tensor operator()(
+        const ttnn::Tensor& input_tensor,
+        std::variant<int, std::array<int, 2>, std::array<int, 3>, std::array<int, 4>> scale_factor,
```

Comment:
```suggestion
        const std::variant<int, std::array<int, 2>, std::array<int, 3>, std::array<int, 4>>& scale_factor,
```

I also think we have tt::tt_metal::Array2D, tt::tt_metal::Array3D, tt::tt_metal::Array4D if you want to use them instead.